.ci/jenkins/Jenkinsfile

```
@@ -0,0 +1,73 @@
+void cancelPrevious() {
+    stage('Cancelling previous') {
+        def buildNumber = env.BUILD_NUMBER as int
+        if (buildNumber > 1) milestone(buildNumber - 1)
+        milestone(buildNumber)
+    }
+}
+
+private Closure runTests(String nodeName) {
+    def ret = {
+        node(nodeName) {
+            stage("${nodeName}") {
+                try {
+                    def scmVars = checkout scm
+                    if (nodeName == 'Linux') {
+                        def dockerImage = 'conanio/ci-functional'
+                        docker.image(dockerImage).inside('--entrypoint=') {
+                            def sourcedir = '/home/conan/'
+                            def pyenvdir = "${sourcedir}.pyenv"
+                            sh(script: "cp -R ./ ${sourcedir}")
+                            sh(script: "chown -R conan ${sourcedir}")
+                            def command = "su - conan -c \"" \
+                                        + "export PYENV_ROOT=${pyenvdir} && " \
+                                        + "export PATH=\"${pyenvdir}/bin:\$PATH\" && " \
+                                        + "pyenv local \$PY36 \$PY38 \$PY39 && " \
+                                        + "python -m pip install tox tox-venv requests && " \
+                                        + "python .ci/last_conan_version.py && " \
+                                        + "python -m tox --verbose\""
+                            sh(script: command)
+                        }
+                    }
+                    else if (nodeName=='Macos') {
+                        def localDir = '/Users/jenkins'
+                        withEnv(['PY36=3.6.12', 'PY38=3.8.6', 'PY39=3.9.0',
+                                "PYENV_ROOT=${localDir}/.pyenv", 
+                                "PATH+EXTRA=${localDir}/.pyenv/shims:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"]) {
+                                def pythonLocation = "${localDir}/.pyenv/versions/\$PY36/bin/python"
+                                def launchCommand = "pyenv local \$PY36 \$PY38 \$PY39 && " \
```

```suggestion
                        withEnv(["PYENV_ROOT=${localDir}/.pyenv", 
                                "PATH+EXTRA=${localDir}/.pyenv/shims:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"]) {
                                def pythonLocation = "${localDir}/.pyenv/versions/3.6.12/bin/python"
                                def launchCommand = "pyenv local 3.6.12 && " \
```

Moreover, if we are using a full path to `pythonLocation`, do we need to activate a local environment at all?

---

Same here for `tox` finding python versions, I suppose.